records/state/state.go

```diff
@@ -103,7 +116,18 @@ func (t *Task) containerIP(src string) string {
 			continue
 		}
 
-		// find the latest docker-inspect label
+		// first try to extract the address from the NetworkInfo structure
+		if len(status.ContainerStatus.NetworkInfos) > 0 {
+			// TODO(CD): handle multiple NetworkInfo objects
+			// TODO(CD): only create A records if the address is IPv4
+			// TODO(CD): create AAAA records if the address is IPv6
+			latestContainerIP = status.ContainerStatus.NetworkInfos[0].IPAddress
```

Comment:
I'm not even sure what Mesos-DNS should do when there are multiple addreses assigned.  I think we'd need conventions for handling additional metadata to give us clues (e.g. labels on NetworkInfo, etc.)